overlord/fdestate/fdemgr.go

```diff
@@ -462,6 +463,24 @@ func (m *FDEManager) GetKeyslots(keyslotRefs []KeyslotRef) (keyslots []Keyslot,
 	return keyslots, missingRefs, nil
 }
 
+// NextKeyID returns a monotonically increasing id that
+// can be used for conflict free ephemeral key names. This
+// can be used in temporary key names during key replacement
+// or re-provisioning.
+//
+// The state needs to be locked by the caller.
+func (m *FDEManager) NextKeyID() (string, error) {
+	var lastKeyID int
+	err := m.state.Get("fde-last-key-id", &lastKeyID)
+	if err != nil && !errors.Is(err, state.ErrNoState) {
+		return "", err
+	}
+	lastKeyID++
+	id := strconv.Itoa(lastKeyID)
+	m.state.Set("fde-last-key-id", lastKeyID)
```

Comment:
Not sure I am a fan of this. What if there is some factory reset. And keys are not cleaned up yet. Maybe there is a conflict.

I would prefer a function that gives a unique name for a specific storage container. And it verifies that we do not have conflict there. Is not it possible to do that?

Reply:
Done, Thank you for the suggestion!